### ISSUE TITLE
immutables.Map() as default arg should not trigger B008

### ISSUE THREAD

**Issue by hukkin** [PyCQA/flake8-bugbear#133]
flake8-bugbear currently allows using the immutable `frozenset()` and `tuple()` types as default args of a function.

Would it be possible to also whitelist `immutables.Map` from https://github.com/MagicStack/immutables ?

**Comment by hukkinj1** [comment]
To elaborate, would be great if the code below would not give me B008 error:
```python
import immutables
from immutables import Map


def this_should_be_fine(mapping = immutables.Map()):
    pass

def this_also(mapping = Map()):
    pass
```

**Comment by cooperlees** [comment]
I'm going to pass this one to our core maintainer @ambv since this immutables.Map is from his MagicStack ...

I personally don't like us using not stdlib objects, but I'm open to this being an extra install maybe? If we find immytables.Map importable, don't fire the line?

**Comment by hukkinj1** [comment]
> I personally don't like us using not stdlib objects

Yeah I can see how this could be a bit of a bad precedent and a scope creep. Still would love having this one exception, as the stdlib does not include an immutable mapping/dict type, and using immutables types (`tuple`, `frozenset`, `immutables.Map`) as default arg makes me not have to write this convention a million times
```python
def func(mapping = None):
    if mapping is None:
        mapping = {}
```

Thanks for considering!

**Comment by escaped** [comment]
How about making the "immutable function calls" configurable via the ini files. Something like [pep8-naming](https://github.com/PyCQA/pep8-naming) did for classmethod/staticmethod. It's not documented, but I can add exceptions to the config file like

```
[flake8]
# pep8-naming
classmethod-decorators =
  classmethod,  # buildin
  validator,  # pydantic
  root_validator,  # pydantic
```

**Comment by hukkin** [comment]
Closing this now that we have https://github.com/PyCQA/flake8-bugbear/pull/204 and https://github.com/PyCQA/flake8-bugbear/pull/144
